schema/2.0.0/corpora_schema.md

```diff
@@ -0,0 +1,257 @@
+
+[Uberon multi-species anatomy ontology]: http://www.obofoundry.org/ontology/uberon.html
+[2021-02-12]: https://github.com/obophenotype/uberon/releases/tag/v2021-02-12
+[uberon.owl]: https://github.com/obophenotype/uberon/blob/v2021-02-12/uberon.owl
+
+### Required Gene Annotations
+
+cellxgene requires ENSEMBL gene identifiers to ensure that all datasets it stores measure the same features and can therefore be integrated.
+
+The following gene annotation dependencies are *pinned* for this version of the schema.
+
+| Organism | Required version | Download |
+|:--|:--|:--|
+| [ENSEMBL (Human)] | Human reference GRCh38 (GENCODE v32/Ensembl 98)] matching [cellranger 2020-A (July 7, 2020) release] | [gencode.v32.primary_assembly.annotation.gtf] |
+| [ENSEMBL (Mouse)] | Mouse reference mm10 (GENCODE vM23/Ensembl 98) matching [cellranger 2020-A (July 7, 2020) release]| [gencode.vM23.primary_assembly.annotation.gtf] |
+|||
+
+[ENSEMBL (Human)]: http://www.ensembl.org/Homo_sapiens/Info/Index
+[gencode.v32.primary_assembly.annotation.gtf]:http://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_32/gencode.v32.primary_assembly.annotation.gtf.gz"
+
+[ENSEMBL (Mouse)]: http://www.ensembl.org/Mus_musculus/Info/Index
+[gencode.vM23.primary_assembly.annotation.gtf]:http://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_mouse/release_M23/gencode.vM23.primary_assembly.annotation.gtf.gz"
+
+[cellranger 2020-A (July 7, 2020) release]: https://support.10xgenomics.com/single-cell-gene-expression/software/release-notes/build
+
+## `obs` (Cell Metadata)
+
+`obs` is a [`pandas.DataFrame`](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.html).
+
+Curators MUST annotate the following columns in the `obs` dataframe:
+
+| Key | Value | Annotator |
+|-|-|-|
+| assay_ontology_term_id | categorical with `str` categories. This MUST be a child of "experimental process". See [EFO:0002694](https://www.ebi.ac.uk/ols/ontologies/efo/terms?short_form=EFO_0002694) for details. If there is not an exact match for the assay, *clarifying text* MAY be appended in parentheses such as `" (sci-plex)"`. | Curator |
+| cell_type_ontology_term_id | categorical with `str` categories. This MUST be a CL term. | Curator |
+| development_stage_ontology_term_id | categorical with `str` categories. If unavailable, this MUST be `"unknown"`. <br><br> If `organism_ontolology_term_id` is `"NCBITaxon:9606"`, this MUST be a HsapDv term.<br><br> If `organism_ontolology_term_id` is `"NCBITaxon:10090"`, this MUST a MmusDv term with the following restrictions: <br><br> **Prenatal stages** MUST be in the range beginning with "MmusDv:0000003" and ending with "MmusDv:0000035". <br><br> **Postnatal stages (1-4 weeks)** MUST be in the range beginning with "MmusDv:0000045" and ending with "MmusDv:0000048". <br><br> **Postnatal stages (after week 4)** MUST be in the range beginning with "MmusDv:0000062" and ending with "MmusDv:0000091". | Curator |
```

Comment:
The reason why we specify it further in mouse is because we wanted curators to stick to months instead of weeks for post-natal stages, but there isn't a 1-month-old term id, in this case they'd need to use the last Theiler stages. I don't believe we have a similar issue in Human, if we there's one we should specify as well.

@brianraymor we could change the text to make it clearer and more concise. I suggest something like:

> If `organism_ontolology_term_id` is `"NCBITaxon:10090"`, this MUST a MmusDv term with the following restrictions:
From the time of conception to 1 month after birth it MUST be a term id associated to "Theiler stages" (e.g. MmusDv:0000003), from 2 months after birth and further it MUST be a term id associated to "X months-old" (e.g. MmusDv:0000077)

Reply:
It might be good to offer human guidance _if_ you don't want year-specific HsapDv terms (like several of the existing datasets have)

Reply:
I think year specific is desired when available

Reply:
Now I'm curious. For humans, you would like the most specific term possible, but for mouse, you'd like a term at a specific level of resolution even when a more specific term is available - why is that?

side note: if you would like the age term when possible for humans, it should be specified here. I think most researchers think "development stage" and will look for the term associated with "adult" rather than dig in and find the most specific term possible.

Reply:
The "weeks" units maxes at 29 weeks (under 6 months old), we decided to go the "months old" unit as it applies to all stages allowing for easier integration and reusability. 

Do you think allowing for weeks may be a better option?

HsapDv seems to be better curated as there's no overlap at different stages and the level of granularity is great. But I think specific guidance would be helpful. Do you think the following adding the following could be useful?

- Embryonic stage : Carnegie stages 1-23 (up to 2 months after conception)
- Fetal development: 9th to 33rd week post-fertilization human stages (3 months after conception and before birth) 
- After birth: X-month-old human stages

Reply:
I admittedly haven't sifted through the development ontologies in-depth (we sectioned them each into 5-7 chronologically sequential terms & then capture the age, in whatever units, separately).
So it was mainly my curiosity, but it makes sense that the differences in the ontologies warrant different approaches.
I do think that additional guidance for human stages would be good (the more guidance you can give on any field, the more likely you'll get the metadata submitted for the same entity/concept)

Reply:
@pablo-gar - can you clarify what specific guidance you want to propose for both hsapdv and mmusdv development_stages?

Reply:
Based on the conversation, I think this is what makes the most sense

1. Add human guidance
2. For clarity change term ids for term labels 

Resulting in a text like this:


> categorical with `str` categories. If unavailable, this MUST be `"unknown"`. <br><br> If `organism_ontolology_term_id` is `"NCBITaxon:9606"` (_Homo sapiens_), this MUST be a HsapDv term with the following restrictions: <br> - Embryonic stage : MUST use term ids associated to "Carnegie stages 1-23" (up to 8 weeks after conception; e.g. HsapDv:0000003) <br> - Fetal development MUST be term ids of "9th to 33rd week post-fertilization human stages" (9 weeks after conception and before birth; e.g HsapDv:0000046) <br> - After birth MUST be term ids of the form "X-month-old human stage" (e.g. HsapDv:0000246) <br><br> If `organism_ontolology_term_id` is `"NCBITaxon:10090"` (_Mus musculus_), this MUST be a MmusDv term with the following restrictions: <br> - From the time of conception to 1 month after birth it MUST be a term id associated to "Theiler stages" (e.g. MmusDv:0000003) <br> - From 2 months after birth and further it MUST be a term id of the form "X months-old" (e.g. MmusDv:0000077)

Reply:
_- After birth MUST be term ids of the form "X-month-old human stage" (e.g. HsapDv:0000246)_

@pablo-gar - HsapDv doesn't have month term ids past 23 months? - http://www.ontobee.org/search?ontology=HSAPDV&keywords=month&submit=Search+terms

Reply:
It was typo, it should have been "year" instead of month, which made me realized there's no "0-year-old human stage", for which we would need to include months

> categorical with `str` categories. If unavailable, this MUST be `"unknown"`. <br><br> If `organism_ontolology_term_id` is `"NCBITaxon:9606"` (_Homo sapiens_), this MUST be a HsapDv term with the following restrictions: <br> - Embryonic stage: it MUST be term ids associated to "Carnegie stages 1-23" (up to 8 weeks after conception; e.g. HsapDv:0000003) <br> - Fetal development MUST be term ids of "9th to 33rd week post-fertilization human stages" (9 weeks after conception and before birth; e.g HsapDv:0000046) <br> - After birth for the first 12 months it MUST be term ids of the form "X-month-old human stage" (e.g. HsapDv:0000174) <br> - After the first 12 months post-birth it MUST be term ids of the form "X-year-old human stage" (e.g. HsapDv:0000246) <br><br> If `organism_ontolology_term_id` is `"NCBITaxon:10090"` (_Mus musculus_), this MUST be a MmusDv term with the following restrictions: <br> - From the time of conception to 1 month after birth it MUST be a term id associated to "Theiler stages" (e.g. MmusDv:0000003) <br> - From 2 months after birth and further it MUST be a term id of the form "X months-old" (e.g. MmusDv:0000077)

Reply:
>After the first 12 months post-birth it MUST be term ids of the form "X-year-old human stage" (e.g. HsapDv:0000246)

I would suggest adding guidance when the exact age cannot be reported for consent reasons. We sometimes see 5-10 year age ranges. And HIPAA regulations state that any age >89 should be grouped as "90+" because there are significantly fewer 90+ individuals that the exact age greatly increases identifiability

Reply:
@pablo-gar - RE

> Fetal development MUST be term ids of "9th to 33rd week post-fertilization human stages" (9 weeks after conception and before birth; e.g HsapDv:0000046)

The range is 9-**38** weeks in the [post-fertilization human stages category](
http://www.ontobee.org/search?ontology=HSAPDV&keywords=post-fertilization&submit=Search+terms).

with the 38th week described as "Fetal stage when the fetus is over 37 weeks of development and until birth."

Was **33** intentional or a typo?

Reply:
Not intentional, thanks for double-checking!

Reply:
Addressed in https://github.com/chanzuckerberg/single-cell-curation/pull/67/commits/138a06464458673c98ba4d2c4618f8eb4e5be477